docs/book/src/user/troubleshooting.md

```diff
@@ -0,0 +1,11 @@
+# Troubleshooting
+
+## Labeling nodes with reserved labels such as `node-role.kubernetes.io` fails with `kubeadm` error during bootstrap.
+
+Reserved labels cannot be set during bootstrapping (for example, by using `node-labels` in `kubeletExtraArgs` when bootstrapping with [CAPBK](https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm)) due to a [restriction](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#noderestriction) on the kubelet self-assigning reserved labels such as `node-roles`.  
```

Comment:
> by using `node-labels`

should this have been:
> by using `node-roles` labels

?

Reply:
> self-assigning reserved labels such as `node-roles`. 

-> `node-role` (singular)

Reply:
@neolit123 please file an issue and/or if you have time, create a follow-up PR. This has already merged. Thanks!